Makefile

```diff
@@ -179,6 +179,9 @@ notice:
 		-noticeTemplate dev-tools/notice/NOTICE.txt.tmpl \
 		-noticeOut NOTICE.txt \
 		-depsOut ""
+	@if [ "$(NPCAP_LICENSE)" == true ]; then \
```

Comment:
Who/what/when will be invoking this target with NPCAP_LICENSE=true? Historically this target is used only when modifying go.mod to regenerate the file that is checked in to the repo.

Do we need a packaging-time target to generate a project (aka beat) specific version of this file that can be included in specific distributions?

Reply:
The logic for this is [here](https://github.com/elastic/beats/pull/29303/commits/3bc0aa7541773d27437425e61c96a951460b4b6c#diff-880b55f7b1bdc3bee186584efd50a8ff82c30bec581155008cf210fe6473092dR2). This ensures that the flag is only set when building the x-pack version on windows.

Reply:
I would expect the zip inspection tests to fail if this is not being run.

Reply:
`make notice` is not run during `mage package`. When the windows zip distributions are built they will pull in the NOTICE.txt from the root of the repo without any changes.

https://github.com/elastic/beats/blob/bc09710f82822904d0fa13a0689f44cdeaaf480c/dev-tools/packaging/packages.yml#L614-L616

If we want to generate a tailored NOTICE.txt for non-OSS zip packages we can do that in mage (rather than make). The place to overwrite the specification that includes the NOTICE.txt would be in this file:

https://github.com/elastic/beats/blob/bc09710f82822904d0fa13a0689f44cdeaaf480c/packetbeat/scripts/mage/package.go#L26-L28

In there we can replace the file with a template concatenates the main NOTICE.txt with the npcap LICENSE file. For example:

```patch
diff --git a/packetbeat/scripts/mage/package.go b/packetbeat/scripts/mage/package.go
index 55f26493cf..810f8951bb 100644
--- a/packetbeat/scripts/mage/package.go
+++ b/packetbeat/scripts/mage/package.go
@@ -18,6 +18,9 @@
 package mage
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/pkg/errors"
 
 	devtools "github.com/elastic/beats/v7/dev-tools/mage"
@@ -48,6 +51,30 @@ func CustomizePackaging() {
 				return devtools.Config(devtools.ReferenceConfigType, c, spec.MustExpand("{{.PackageDir}}"))
 			},
 		}
+		npcapNoticeTxt = devtools.PackageFile{
+			Mode:   0o644,
+			Source: "{{.PackageDir}}/NOTICE.txt",
+			Dep: func(spec devtools.PackageSpec) error {
+				repo, err := devtools.GetProjectRepoInfo()
+				if err != nil {
+					return err
+				}
+
+				notice, err := ioutil.ReadFile(filepath.Join(repo.RootDir, "NOTICE.txt"))
+				if err != nil {
+					return err
+				}
+				notice = append(notice, '\n')
+
+				license, err := ioutil.ReadFile(devtools.XPackBeatDir("npcap/installer/LICENSE"))
+				if err != nil {
+					return err
+				}
+				notice = append(notice, license...)
+
+				return ioutil.WriteFile(spec.MustExpand("{{.PackageDir}}/NOTICE.txt"), notice, 0o644)
+			},
+		}
 	)
 
 	for _, args := range devtools.Packages {
@@ -56,6 +83,9 @@ func CustomizePackaging() {
 			case devtools.TarGz, devtools.Zip:
 				args.Spec.ReplaceFile("{{.BeatName}}.yml", configYml)
 				args.Spec.ReplaceFile("{{.BeatName}}.reference.yml", referenceConfigYml)
+				if devtools.BeatLicense == "Elastic License" && args.OS == "windows" {
+					args.Spec.ReplaceFile("NOTICE.txt", npcapNoticeTxt)
+				}
 			case devtools.Deb, devtools.RPM, devtools.DMG:
 				args.Spec.ReplaceFile("/etc/{{.BeatName}}/{{.BeatName}}.yml", configYml)
 				args.Spec.ReplaceFile("/etc/{{.BeatName}}/{{.BeatName}}.reference.yml", referenceConfigYml)
```

Reply:
Yeah, I started out working with mage (and still have that work stashed) until I saw that the notice is created by a standalone program.

I'll do this, but now I'm concerned because the tests don't fail.

Reply:
There's something not quite right with this. It does do the work, but the file that ends up in the package is the original. Happily, the tests catch that now.